src/ProjectTemplates/Web.ProjectTemplates/content/SignalR-CSharp/wwwroot/js/chat.js

```diff
@@ -0,0 +1,60 @@
+var chatConnection = new signalR.HubConnectionBuilder().withUrl("/chat").build();
+
+chatConnection.on("Send", function (message) {
+    var li = document.createElement("li");
+    li.textContent = message;
+    document.getElementById("messagesList").appendChild(li);
+});
+
+document.getElementById("groupmsg").addEventListener("click", async (event) => {
+    var userName = document.getElementById("user-name").value;
+    var groupName = document.getElementById("group-name").value;
+    var groupMsg = document.getElementById("group-message-text").value;
+    try {
+        await chatConnection.invoke("SendMessageToGroup", userName, groupName, groupMsg);
+    }
+    catch (e) {
+        console.error(e.toString());
+    }
+    event.preventDefault();
+});
+
+document.getElementById("join-group").addEventListener("click", async (event) => {
+    var userName = document.getElementById("user-name").value;
+    var groupName = document.getElementById("group-name").value;
+    try {
+        await chatConnection.invoke("AddToGroup", userName, groupName);
+    }
+    catch (e) {
+        console.error(e.toString());
+    }
+    event.preventDefault();
+});
+
+document.getElementById("leave-group").addEventListener("click", async (event) => {
+    var userName = document.getElementById("user-name").value;
+    var groupName = document.getElementById("group-name").value;
+    try {
+        await chatConnection.invoke("RemoveFromGroup", userName, groupName);
+    }
+    catch (e) {
+        console.error(e.toString());
+    }
+    event.preventDefault();
+});
+
+async function start() {
+    try {
+        await chatConnection.start();
+        console.log("connected");
+    } catch (err) {
+        console.log(err);
+        setTimeout(() => start(), 5000);
```

Comment:
turn on automatic reconnect?

Reply:
done, sir! good idea. thanks again to @halter73